common/djangoapps/student/views/dashboard.py

```diff
@@ -225,6 +225,53 @@ def get_course_enrollments(user, org_whitelist, org_blacklist):
             yield enrollment
 
 
+def get_filtered_course_entitlements(user, org_whitelist, org_blacklist):
+    """
+    Given a user, return a filtered set of his or her course enrollments.
+
+    Arguments:
+
+    Returns:
+        generator[CourseEntitlement]: a sequence of enrollments to be displayed
+        on the user's dashboard.
+    """
+    course_entitlement_available_sessions = {}
+    unfulfilled_entitlement_pseudo_sessions = {}
+    course_entitlements = list(CourseEntitlement.get_active_entitlements_for_user(user))
+    filtered_entitlements = []
+    pseudo_session = None
+    course_key_str = ''
+
+    for course_entitlement in course_entitlements:
+        course_entitlement.update_expired_at()
+        available_runs = get_visible_sessions_for_entitlement(course_entitlement)
+
+        if not course_entitlement.enrollment_course_run or not available_runs:
```

Comment:
should this be `if not (course_entitlement.enrollment_course_run or available_runs):` ?

Reply:
or I suppose it could also be `if not course_entitlement.enrollment_course_run and not available_runs:`